File: check_models.ts
import 'dotenv/config';

// Hack para garantir que o fetch funcione em qualquer versão do Node
declare const fetch: any;

async function listGoogleModels() {
    const apiKey = process.env.GEMINI_API_KEY;

    if (!apiKey) {
        console.error("❌ ERRO: GEMINI_API_KEY não encontrada no .env");
        return;
    }

    console.log("📡 Perguntando ao Google: 'O que eu posso usar?'...\n");

    try {
        // Chama a API direta do Google para listar modelos (GET /models)
        const response = await fetch(`https://generativelanguage.googleapis.com/v1beta/models?key=${apiKey}`);
        const data = await response.json();

        if (data.error) {
            console.error(`❌ Erro do Google: ${data.error.message}`);
            return;
        }

        if (!data.models || data.models.length === 0) {
            console.log("⚠️ O Google respondeu, mas a lista de modelos veio vazia.");
            return;
        }

        console.log("✅ Lista Oficial de Modelos Disponíveis para sua Chave:");
        console.log("======================================================");

        // Filtra e exibe
        data.models.forEach((model: any) => {
            // Mostra apenas modelos que geram texto (ignora embeddings se quiser limpar, 
            // mas vou deixar tudo já que vc pediu "o que o google responder")
            
            const capabilities = model.supportedGenerationMethods.join(", ");
            console.log(`📦 NOME: ${model.name.replace('models/', '')}`); // Remove o prefixo 'models/' pra ficar limpo
        });

    } catch (error) {
        console.error("🔥 Erro fatal ao conectar:", error);
    }
}

listGoogleModels();

File: python_codes/check_models.py
import google.generativeai as genai
import os
from dotenv import load_dotenv
from pathlib import Path

env_path = Path(__file__).parent.parent / '.env'
load_dotenv(dotenv_path=env_path)

api_key = os.getenv("GEMINI_API_KEY")
print(f"🔑 Testando chave: {api_key[:5]}...{api_key[-5:]}")

genai.configure(api_key=api_key)

print("\n📡 Perguntando ao Google quais modelos você pode usar...")

try:
    found = False
    for m in genai.list_models():
        if 'generateContent' in m.supported_generation_methods:
            print(f"✅ MODELO DISPONÍVEL: {m.name}")
            found = True
    
    if not found:
        print("❌ Nenhum modelo de texto encontrado.")

except Exception as e:
    print(f"🔥 Erro fatal ao conectar: {e}")